src/yuzu/main.cpp

```diff
@@ -4845,6 +4855,27 @@ bool GMainWindow::CheckFirmwarePresence() {
     return true;
 }
 
+void GMainWindow::SetFirmwareVersion() {
+    Service::Set::FirmwareVersionFormat firmware_data{};
+    const auto result = Service::Set::GetFirmwareVersionImpl(
+        firmware_data, *system, Service::Set::GetFirmwareVersionType::Version2);
+
+    firmware_label->setVisible(true);
+
+    if (result.IsError()) {
+        LOG_INFO(Frontend, "Installed Firmware: No firmware available");
+        firmware_label->setText(tr("No firmware"));
```

Comment:
Maybe consider hiding the label if there is no firmware?

Also, you are missing a return statement
```suggestion
        LOG_INFO(Frontend, "No firmware available");
        firmware_label->setText(tr("No firmware"));
        return;
```